.github/workflows/go.yml

```diff
@@ -0,0 +1,43 @@
+name: Go
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+
+  build:
+    name: Build
+    runs-on: macos-latest
```

Comment:
Any reason why run on macOS? Is it going faster/slower or what?

Reply:
we have our machines as macOS on which we are presently building, therefore i chose to keep it macos. I guess if needed, we can for sure switch to ubuntu